File: skills/agy-frontend/tests/test_verify_static_frontend.ts
import assert from "node:assert/strict";
import { spawnSync } from "node:child_process";
import { mkdirSync, mkdtempSync, writeFileSync } from "node:fs";
import { tmpdir } from "node:os";
import path from "node:path";
import test from "node:test";

const SKILL_ROOT = path.resolve(import.meta.dirname, "..");
const VERIFY_SCRIPT = path.join(SKILL_ROOT, "scripts/verify-static-frontend.ts");

function runVerify(siteRoot: string, envOverrides: Record<string, string> = {}) {
  return spawnSync(process.execPath, ["--experimental-strip-types", VERIFY_SCRIPT, siteRoot], {
    encoding: "utf8",
    env: {
      ...process.env,
      VERIFY_BROWSER: "0",
      ...envOverrides,
    },
  });
}

test("video minimum fails when referenced video is missing", () => {
  const site = mkdtempSync(path.join(tmpdir(), "agy-missing-video-"));
  writeFileSync(
    path.join(site, "index.html"),
    '<!doctype html><title>Video check</title><video src="media/missing-loop.mp4"></video>',
    "utf8",
  );

  const result = runVerify(site, { ASSET_MIN_VIDEOS: "1" });

  assert.notEqual(result.status, 0, result.stdout + result.stderr);
  assert.match(result.stdout + result.stderr, /missing/);
  assert.match(result.stdout + result.stderr, /ASSET_MIN_VIDEOS=1/);
});

test("video minimum passes when referenced video exists", () => {
  const site = mkdtempSync(path.join(tmpdir(), "agy-existing-video-"));
  const media = path.join(site, "media");
  mkdirSync(media);
  writeFileSync(path.join(media, "hero-loop.mp4"), "not-empty-video-placeholder");
  writeFileSync(
    path.join(site, "index.html"),
    '<!doctype html><title>Video check</title><video src="media/hero-loop.mp4"></video>',
    "utf8",
  );

  const result = runVerify(site, { ASSET_MIN_VIDEOS: "1" });

  assert.equal(result.status, 0, result.stdout + result.stderr);
  assert.match(result.stdout, /local video assets referenced: 1/);
});

File: skills/agy-frontend/tests/test_verify_static_frontend.py
from __future__ import annotations

import os
import subprocess
import tempfile
import unittest
from pathlib import Path


SKILL_ROOT = Path(__file__).resolve().parents[1]
VERIFY_SCRIPT = SKILL_ROOT / "scripts" / "verify-static-frontend.sh"


def run_verify(site_root: Path, **env_overrides: str) -> subprocess.CompletedProcess[str]:
    env = os.environ.copy()
    env.update(env_overrides)
    env.setdefault("VERIFY_BROWSER", "0")
    return subprocess.run(
        [str(VERIFY_SCRIPT), str(site_root)],
        text=True,
        stdout=subprocess.PIPE,
        stderr=subprocess.STDOUT,
        env=env,
        check=False,
    )


class VerifyStaticFrontendVideoAssetsTest(unittest.TestCase):
    def test_video_minimum_fails_when_referenced_video_is_missing(self) -> None:
        with tempfile.TemporaryDirectory() as temp_dir:
            site = Path(temp_dir)
            (site / "index.html").write_text(
                "<!doctype html><title>Video check</title>"
                '<video src="media/missing-loop.mp4"></video>',
                encoding="utf-8",
            )

            result = run_verify(site, ASSET_MIN_VIDEOS="1")

            self.assertNotEqual(result.returncode, 0, result.stdout)
            self.assertIn("missing", result.stdout)
            self.assertIn("ASSET_MIN_VIDEOS=1", result.stdout)

    def test_video_minimum_passes_when_referenced_video_exists(self) -> None:
        with tempfile.TemporaryDirectory() as temp_dir:
            site = Path(temp_dir)
            media = site / "media"
            media.mkdir()
            (media / "hero-loop.mp4").write_bytes(b"not-empty-video-placeholder")
            (site / "index.html").write_text(
                "<!doctype html><title>Video check</title>"
                '<video src="media/hero-loop.mp4"></video>',
                encoding="utf-8",
            )

            result = run_verify(site, ASSET_MIN_VIDEOS="1")

            self.assertEqual(result.returncode, 0, result.stdout)
            self.assertIn("local video assets referenced: 1", result.stdout)


if __name__ == "__main__":
    unittest.main()
